sources/whippet/src/child.rs

```diff
@@ -0,0 +1,125 @@
+//! Child process management for dbus-broker.
+//!
+//! This module provides the [`BrokerChild`] which manages the lifecycle of the spawned
+//! dbus-broker process, including signal handling and resource cleanup.
+
+use crate::error::{self, Result};
+use snafu::{OptionExt, ResultExt};
+use std::os::unix::io::AsRawFd;
+use std::os::unix::net::{UnixDatagram, UnixStream};
+use std::process::Stdio;
+use tokio::process::{Child, Command};
+use tokio::signal;
+
+/// Hardcoded path to machine-id file in Bottlerocket
+const MACHINE_ID_PATH: &str = "/etc/machine-id";
+
+/// Default limits from dbus-launcher
+/// See: https://github.com/bus1/dbus-broker/blob/b0db0890d1254477cf832e5f9f0a798360c80fd9/src/launch/launcher.c#L35
+const DEFAULT_MAX_OUTGOING_BYTES: u64 = 8 * 1024 * 1024;
+const DEFAULT_MAX_OUTGOING_UNIX_FDS: u64 = 64;
+const DEFAULT_MAX_CONNECTIONS_PER_USER: u64 = 64;
+const DEFAULT_MAX_MATCH_RULES_PER_CONNECTION: u64 = 256;
+
+/// Calculated limits (per-user = per-connection × max-connections)
+const MAX_BYTES: u64 = DEFAULT_MAX_CONNECTIONS_PER_USER * DEFAULT_MAX_OUTGOING_BYTES;
+const MAX_FDS: u64 = DEFAULT_MAX_CONNECTIONS_PER_USER * DEFAULT_MAX_OUTGOING_UNIX_FDS;
+const MAX_MATCHES: u64 = DEFAULT_MAX_CONNECTIONS_PER_USER * DEFAULT_MAX_MATCH_RULES_PER_CONNECTION;
+
+/// Manages the lifecycle of the spawned child process
+#[derive(Debug)]
+pub(crate) struct BrokerChild {
+    broker_path: String,
+    journal_socket: UnixDatagram,
+    controller_socket: UnixStream,
+    /// Optinal child as its initialization is delayed until `run` is called
+    child: Option<Child>,
+}
+
+impl BrokerChild {
+    /// Returns a new instance of the `BrokerChild`, taking ownership of the provided sockets
+    pub(crate) fn new(
+        broker_path: &str,
+        controller_socket: UnixStream,
+        journal_socket: UnixDatagram,
+    ) -> Result<Self> {
+        Ok(Self {
+            broker_path: broker_path.to_string(),
+            journal_socket,
+            controller_socket,
+            child: None,
+        })
+    }
+
+    /// Forks the dbus-broker with the parameters configured for the child
+    pub(crate) fn run(&mut self) -> Result<()> {
+        debug!("Reading machine ID from {}", MACHINE_ID_PATH);
+        let machine_id = std::fs::read_to_string(MACHINE_ID_PATH)
+            .context(error::MachineIdSnafu {
+                path: MACHINE_ID_PATH,
+            })?
+            .trim()
+            .to_string();
+        info!("Machine ID: {}", machine_id);
```

Comment:
I wouldn't elevate this to `info`:
```suggestion
        debug!("Machine ID: {}", machine_id);
```